hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -127,6 +127,27 @@ public class HoodieTableConfig extends HoodieConfig {
       .withDocumentation("Columns used to uniquely identify the table. Concatenated values of these fields are used as "
           + " the record key component of HoodieKey.");
 
+  public static final ConfigProperty<Boolean> CDC_ENABLED = ConfigProperty
+      .key("hoodie.table.cdc.enabled")
+      .defaultValue(false)
+      .withDocumentation("When enable, persist the change data if necessary, and can be queried as a CDC query mode.");
+
+  // op and key
+  public static final String CDC_SUPPLEMENTAL_LOGGING_MODE_MINI = "op_key";
+  public static final String CDC_SUPPLEMENTAL_LOGGING_MODE_WITH_BEFORE = "cdc_data_before";
+  public static final String CDC_SUPPLEMENTAL_LOGGING_MODE_WITH_BEFORE_AFTER = "cdc_data_before_after";
+
+  public static final ConfigProperty<String> CDC_SUPPLEMENTAL_LOGGING_MODE = ConfigProperty
+      .key("hoodie.table.cdc.supplemental.logging.mode")
+      .defaultValue(CDC_SUPPLEMENTAL_LOGGING_MODE_WITH_BEFORE_AFTER)
```

Comment:
can we make default KEY_OP instead of full logging? we want to minimize storage out of the box and only log as needed

Reply:
ok.